osu.Game/Database/RealmExtensions.cs

```diff
@@ -8,18 +8,45 @@ namespace osu.Game.Database
 {
     public static class RealmExtensions
     {
+        /// <summary>
+        /// Perform a write operation against the provided realm instance.
+        /// </summary>
+        /// <remarks>
+        /// This will automatically start a transaction if not already in one.
+        /// </remarks>
+        /// <param name="realm">The realm to operate on.</param>
+        /// <param name="function">The write operation to run.</param>
         public static void Write(this Realm realm, Action<Realm> function)
         {
-            using var transaction = realm.BeginWrite();
+            Transaction? transaction = null;
+
+            if (!realm.IsInTransaction)
+                transaction = realm.BeginWrite();
+
             function(realm);
-            transaction.Commit();
+
+            transaction?.Commit();
```

Comment:
Does the transaction require disposal? Previous code did.

Reply:
Yeah, to perform rollback. Slipped my mind.